lib/charms/observability_libs/v0/metrics_endpoint_discovery.py

```diff
@@ -126,35 +127,40 @@ def __init__(self, charm: CharmBase, labels: Dict[str, Iterable]):
         self._observer_pid = 0
 
         self._labels = labels
+        self.start_observer()
 
     def start_observer(self):
         """Start the metrics endpoint observer running in a new process."""
         self.stop_observer()
 
-        logging.info("Starting metrics endpoint observer process")
-
         # We need to trick Juju into thinking that we are not running
         # in a hook context, as Juju will disallow use of juju-run.
         new_env = os.environ.copy()
         if "JUJU_CONTEXT_ID" in new_env:
             new_env.pop("JUJU_CONTEXT_ID")
 
+        tool_prefix = f"/var/lib/juju/tools/{self.unit_tag}"
```

Comment:
I thought we didn't need to support juju2. Do we?

Reply:
_We_ do not, but KubeFlow is staying on 2.9 because it's LTS 🤷

Reply:
We do point to `/usr/bin/juju-exec` in this patch if the other doesn't exist. I don't have a 2.9 environment to test against here, and the assumption was that the path in [current version of the library](https://github.com/canonical/observability-libs/blob/main/lib/charms/observability_libs/v0/metrics_endpoint_discovery.py#L147) was correct. If you can verify that it actually is `/usr/bin/juju-run` on 2.9, we should also change that.

Reply:
Confirmed :point_up: with 2.9.38.

Reply:
Done.